core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -321,7 +321,7 @@ private[spark] class Executor(
           logInfo(s"Executor killed $taskName (TID $taskId)")
           execBackend.statusUpdate(taskId, TaskState.KILLED, ser.serialize(TaskKilled))
 
-        case cDE: CommitDeniedException =>
+        case CausedBy(cDE: CommitDeniedException) =>
```

Comment:
This also only handles 1 level, note. If it can be reused to improve other code that would be convincing.